File: resize_icon.js
const fs = require('fs');
const { createCanvas, loadImage } = require('canvas');

async function resizeIcon() {
  const img = await loadImage('docs/image.png');
  console.log(`Original size: ${img.width}x${img.height}`);

  const sizes = {
    'mdpi': 48,
    'hdpi': 72,
    'xhdpi': 96,
    'xxhdpi': 144,
    'xxxhdpi': 192
  };

  for (const [density, size] of Object.entries(sizes)) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // 고품질 리샘플링
    ctx.imageSmoothingEnabled = true;
    ctx.imageSmoothingQuality = 'high';
    ctx.drawImage(img, 0, 0, size, size);
    
    const buffer = canvas.toBuffer('image/png');
    const dir = `app/src/main/res/mipmap-${density}`;
    
    fs.writeFileSync(`${dir}/ic_launcher.png`, buffer);
    fs.writeFileSync(`${dir}/ic_launcher_round.png`, buffer);
    
    console.log(`Created ${density}: ${size}x${size}`);
  }
  
  console.log('All icons created!');
}

resizeIcon().catch(console.error);

File: resize_icon.py
from PIL import Image
import os

# 원본 이미지 열기
source = Image.open('docs/image.png')
print(f'Original size: {source.size}')

# 각 해상도별 크기 정의
sizes = {
    'mdpi': 48,
    'hdpi': 72,
    'xhdpi': 96,
    'xxhdpi': 144,
    'xxxhdpi': 192
}

# 각 밀도별로 리사이즈
for density, size in sizes.items():
    # 고품질 리샘플링으로 리사이즈
    resized = source.resize((size, size), Image.Resampling.LANCZOS)
    
    output_dir = f'app/src/main/res/mipmap-{density}'
    
    # ic_launcher.png
    resized.save(f'{output_dir}/ic_launcher.png', 'PNG', optimize=True)
    # ic_launcher_round.png
    resized.save(f'{output_dir}/ic_launcher_round.png', 'PNG', optimize=True)
    
    print(f'Created {density}: {size}x{size}')

print('All icons created!')
